### DISCUSSION TITLE
Conflict of two Query Loop Variation blocks on the same page.

### DISCUSSION THREAD

**Discussion by realthemes** [WordPress/gutenberg#62629] [Q&A] [answered]
Hello,

I found an article on the WordPress Block Development portal about creating a Query Loop Variation block. I followed that guide, and my two separate variation plugins seem to work individually, but not when both are used on the same page. When I add the second block, it starts to give me unexpected results for post sorting. However, when I disable one of the variations, everything works fine.

Can someone tell me where I'm wrong?
I can provide any part of my plugins to check, like the `pre_render_block` and `rest_post_query` filters that handle the post queries and sorting logic.

**Comment by skorasaurus** [comment]
This sounds like it might be the result of duplicate query IDs.  

There is an existing issue open about this https://github.com/WordPress/gutenberg/issues/55823 but there is a manual way to work around it ; by manually changing the queryid of one of either loops in the code editor, on the page; you will have to do this for every page or template that has more than 1 query loop on it.

**Reply by danylfilatov** [reply]
godsend github reply. + also fixed my issue with broken post template element styles when paginating.

**Comment by realthemes** [comment]
@skorasaurus Thank you, now I see, but I'm also just check my IDs of Query Loops in Code Editor, and they seem different, "queryId":2 and "queryId":6. I'm also change them to something random and this doesn't help me. Any other suggestions, may be create one plugin for two Query Loop variations, like 2in1?

**Comment by realthemes** [comment, marked as answer]
@skorasaurus My IDs are not duplicating, so in this case, the problem was that I didn't check the `queryId` of the block before adding a filter on the `query_loop_block_query_vars` hook. When I did that, the problem was gone. Thank you for trying to help.

**Reply by skorasaurus** [reply]
Glad to hear that you had it worked out and solved. :)
